pkg/controller/injector/k8s_binding_injector.go

```diff
@@ -58,13 +61,19 @@ func (b *k8sBindingInjector) Inject(binding *servicecatalog.Binding, cred *broke
 			Name:      binding.Name,
 			Namespace: binding.Namespace,
 		},
-		Data: map[string][]byte{
-			"hostname": []byte(cred.Hostname),
-			"port":     []byte(cred.Port),
-			"username": []byte(cred.Username),
-			"password": []byte(cred.Password),
-		},
 	}
+
+	// For each item in the cred just serialize its value into JSON and
+	// save it in the secret
+	for k, v := range *cred {
+		data, err := json.Marshal(v)
```

Comment:
Base64 is how byte arrays are encoded in json. Json/yaml encoder does the encoding.